### PR TITLE
Add create_pool LWRP

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

**Comment by vuntz** [comment]
TBH, I feel like this should be a new LWRP, not something added to the existing LWRP.

**Comment by tboerger** [comment]
A new LWRP sounds not bad, but i've got not a strong opinion on that.

File: chef/cookbooks/ceph/resources/client.rb [hunk 1]
```diff
@@ -11,6 +11,10 @@
 # defaults to client.#{name}.#{hostname}
 attribute :keyname, :kind_of => String
 
+# what pool should be created in the ceph cluster
+attribute :pool_name, :kind_of => String
+attribute :pg_num, :kind_of => [Integer, String]
```
Comment (dirkmueller) [inline chef/cookbooks/ceph/resources/client.rb]: I don't think it makes a lot of sense to accept strings here..

File: chef/cookbooks/ceph/providers/client.rb [hunk 2]
```diff
@@ -72,3 +84,11 @@ def auth_set_key(keyname, caps)
   end
   get_or_create.error!
 end
+
+def create_pool(pool_name, pg_num)
+  Chef::Log.info("Creating ceph pool '#{pool_name}' '#{pg_num}'")
```
Comment (dirkmueller) [inline chef/cookbooks/ceph/providers/client.rb]: IMHO pg_num should be an optional parameter here, and it should default to something sensible (depending on the pool size ?)

**Comment by dirkmueller** [comment]
@vuntz when we move it to "OpenStack" Barclamp, is that then a new LWRP for you? ;)

**Comment by rhafer** [comment]
This needs to be rebased. And I somehow agree with @vuntz that would better fit into a separate chef-resource (```ceph_pool```?) than adding it to the ```chef_client``` resource.

**Comment by mjura** [comment]
I've created new `ceph_pool` resource, it was tested and works fine.

File: chef/cookbooks/ceph/recipes/glance.rb [hunk 3]
```diff
@@ -19,6 +19,7 @@
   mode 0640
 end
 
-execute "create new pool #{glance_pool}" do
-  command "ceph osd pool create #{glance_pool} 64"
+ceph_pool glance_pool do
+  pool_name glance_pool
+  pg_num 128
```
Comment (rhafer) [inline chef/cookbooks/ceph/recipes/glance.rb]: My ceph knowledge isn't that good, so maybe my question is stupid. But why the change from 64 to 128? Was that indented or just an accident?

If it was indented, mentioning the reasons it the commit message would be good. In general I think it is a good pratice to have separate commits for refactoring (the change from the execute resource to the ceph_pool resource) and functional changes (changing the pg_num default).

File: chef/cookbooks/ceph/providers/pool.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+def whyrun_supported?
+  true
```
Comment (rhafer) [inline chef/cookbooks/ceph/providers/pool.rb]: Are you sure? 
According to http://dougireton.com/blog/2013/01/07/creating-an-lwrp-part-2/ make a resource supporitng ```whyrun``` requires adding some ```converge_by``` blocks. I don't see any. So maybe you just want to remove this.

File: chef/cookbooks/ceph/recipes/glance.rb (hunk 3 above)
Comment (mjura) [inline chef/cookbooks/ceph/recipes/glance.rb]: Ups, I did it by accidentm, this value will be calculated automatically in this https://github.com/crowbar/barclamp-ceph/pull/92

File: chef/cookbooks/ceph/providers/pool.rb (hunk 4 above)
Comment (mjura) [inline chef/cookbooks/ceph/providers/pool.rb]: You are right, I was wondering for what reason we need this in `ceph_client` LWRP too.

**Comment by rhafer** [comment]
LGTM. @jsuchome Would you mind having a look a this?

File: chef/cookbooks/ceph/resources/pool.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+actions :create
+default_action :create
+
+attribute :name, :kind_of => String, :name_attribute => true
+
+# Whether we have ceph cluster deployed without crowbar
```
Comment (jsuchome) [inline chef/cookbooks/ceph/resources/pool.rb]: Is this comment related to :ceph_conf attribute?

**Comment by jsuchome** [comment]
In general, LGTM

File: chef/cookbooks/ceph/resources/pool.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+# Whether we have ceph cluster deployed with non standard configuration
+# and ceph.conf and admin keyring is placed somewhere else than in /etc/ceph
+# This can happened when we want to use external ceph cluster deployed 
+# without crowbar
+attribute :ceph_conf, :kind_of => String, :default => '/etc/ceph/ceph.conf'
+attribute :admin_keyring, :kind_of => String, :default => '/etc/ceph/ceph.client.admin.keyring'
+
+# what pool should be created in the ceph cluster
+attribute :pool_name, :kind_of => String
```
Comment (vuntz) [inline chef/cookbooks/ceph/resources/pool.rb]: Why do we have both pool_name and name?

File: chef/cookbooks/ceph/providers/pool.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  create_pool(ceph_conf, admin_keyring, pool_name, pg_num)
+end
+
+def load_current_resource
+  @current_resource = Chef::Resource::CephPool.new(@new_resource.name)
+end
+
+def create_pool(ceph_conf, admin_keyring, pool_name, pg_num)
+  Chef::Log.info("Creating ceph pool '#{pool_name}' with pg number '#{pg_num}'")
+  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} osd pool create '#{pool_name}' '#{pg_num}'"
```
Comment (vuntz) [inline chef/cookbooks/ceph/providers/pool.rb]: pool_name should be escaped for shell usage...

File: chef/cookbooks/ceph/resources/pool.rb (hunk 6 above)
Comment (mjura) [inline chef/cookbooks/ceph/resources/pool.rb]: It was fixed in PR https://github.com/crowbar/barclamp-ceph/pull/140

File: chef/cookbooks/ceph/providers/pool.rb (hunk 7 above)
Comment (mjura) [inline chef/cookbooks/ceph/providers/pool.rb]: This was addressed in PR https://github.com/crowbar/barclamp-ceph/pull/144
